core/runtime/src/main/java/io/quarkus/runtime/annotations/ConfigItem.java

```diff
@@ -61,4 +61,15 @@
      * @return the default value documentation
      */
     String defaultValueDocumentation() default "";
+
+    /**
+     * To ensure backward compatibility with earlier names when a property is renamed.
+     * Will be ignored if the property exists with current name else this will be used to resolve the value and
+     * a warning will be printed that the given property key is deprecated.
+     * If properties are present for multiple deprecated names, then the first one(in the below array) to get resolved
+     * will be used and rest will be ignored.
+     *
+     * @return deprecated names for the Item.
+     */
+    String[] deprecatedNames() default {};
```

Comment:
I think I would go with the separate annotations as @dmlloyd suggested it.

The main reason is that once we want to remove the deprecated names, we can just drop the annotation.

I think I would also make the annotation repeatable instead of using an array. The main reason is that I can envision we could need some converter to convert the old value to the new value and that would be more futureproof.

Does it make sense? (Better wait for consensus before changing that as it might just be a crazy idea :))

/cc @dmlloyd

Reply:
Thanks for your feedback @gsmet . I didn't create new annotation because I thought that the new additional won't be useful if not used along with ConfigItem. That's why added a new field in current annotation. And regarding making the annotation repeatable, I'm not sure If I understand the use case because instead of changing the type of field, a new property would be better option. 
As you've suggested, Let's wait for suggestions from other people. Please let me know If I've missed something. :)

Reply:
I agree with @gsmet that an annotation per alias makes more sense - something like `@OldName` - so we can potentially customize each old name if necessary.